common/lib/api-client/auth.js

```diff
@@ -56,25 +57,28 @@ Auth.prototype = {
 
     debug('AUTH REQUEST', url)
 
-    // get metrics instrumentation
-    const clientInstrumentation = clientMetrics.start({
+    // start request timer
+    const authTimer = timer()
+    const reqLabels = {
       url,
       method: 'POST',
-    })
+    }
 
     return axios
       .post(this.config.authUrl, data, config)
       .then(response => {
         debug('AUTH SUCCESS', url)
         // record successful request
-        clientMetrics.stop(clientInstrumentation, response)
+        const duration = authTimer()
```

Comment:
It might just be me, but it would feel more readable if the timer had more verbose method to call to start and get the current time:

```
timer.start()
timer.timeElapsed()
```

Reply:
I am following the likes of things like process.hrtime and indeed promClient's historgram and summary timed functions

ie.

```
end = start()
...
duration = end()
```

rather than, say,

```
timer = new Timer()
timer.start()
...
timer.end()
duration = timer.report()
```